app/static/admin/js/pages/gen_form/form-donation.js

```diff
@@ -1,174 +1,256 @@
+        // [1] donor name not present
+        if (this == dom.button.addNew && !util.isDonorNamePresent()) {
+            util.enterDonorName();
+            return;
+        }
 
-          _this.div.taxReceiptNo.hidden = true;
-          util.scrollTo(_this.input.date);
-          return;
+        // [2] create new donation
+        if (this == dom.button.addNew) {
+            util.emptyAllFields(dom.input);             // empty donation input fields
+            item.clearItemView();                       // clear items table and form
+            util.setButton(dom.button, "new");          // show appropriate button for new data
+            dom.div.form.hidden         = false;        // make sure form is shown
+            dom.div.taxReceiptNo.hidden = true;         // tax_receipt_no field is hidden
+            util.scrollTo(dom.input.date);              // scroll to input date
+            dom.input.donorId.value = store.donor_id;   // set donor id into a form field
+            return;
         }
 
-        // [1] event when form needs to be closed
-        if (this == _this.button.cancel || !data) {
-          _this.div.form.hidden = true;
-          _this.div.header.hidden = true;
-
-          util.emptyAllFields(_this.input);
-          item.printItemList(null);
-          util.setButton(_this.button, null);
-          return;
-        } else {
-          // [3] event to set form with data
-          util.setButton(_this.button, "existing");
-
-          _this.div.taxReceiptNo.hidden = false;
-          // _this.div.header.hidden = false;
-          // _this.div.header.innerText = data.tax_receipt_no;
-
-          _this.input.taxReceiptNo.value = data.tax_receipt_no || "";
-          _this.input.date.value = data.donate_date || "";
-          _this.input.isVerified.checked =
-            data.verified.toUpperCase() == "TRUE";
-          _this.input.pickUpPostalCode.value = data.pick_up || "";
+        // [3] event when form needs to be closed
+        if (this == dom.button.cancel || !data) {
+            dom.div.form.hidden     = true;             // hide form
+            // dom.div.header.hidden   = true;          // header no longer used
+
+            util.emptyAllFields(dom.input);             // clear out the input fields
+            item.clearItemView();                       // clear items table
+            util.setButton(dom.button, null);           // set button to a default
+            return;
         }
 
-        _this.div.form.hidden = false;
-      };
-    }.call(this);
+        // set form iff data.tax_receipt_no match with the last selected donation.
+        if (!util.check('tax_receipt_no', data.tax_receipt_no)) return;
+
+        // [4] event to set form with data
+        util.setButton(dom.button, "existing");
+
+        // set input fields with data
+        dom.input.donorId.value             = data.donor_id_id;
+        dom.input.taxReceiptNo.value        = data.tax_receipt_no || "tax receipt not given";
```

Comment:
What is the "tax receipt not given" case for? Shouldn't it be set to empty string or undefined?